### PR TITLE
feat(theme): add $icon-interactive token

### LINKED ISSUE #13217: Experimental tile: Add a new color token for the icon
**Background:** This work started with icons placed at the bottom right in tiles. Using the `$interactive` token does not pass the accessibility requirements in dark themes. We wanted to create a new color token for this icon. This can also apply to other elements in the future. 

Token name: `$icon-interactive`
- White and Gray 10 theme: Blue 60 #0f62fe
- Gray 90 and Gray 100 theme: White #ffffff

- [ ] Develop the token (developer)

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/13217

Adds a new `$icon-interactive` token

White and Gray 10 theme: Blue 60 `#0f62fe`

Gray 90 and Gray 100 theme: White `#ffffff`

#### Changelog

**New**

- `$icon-interactive` added to all themes

**Changed**

- Updated snapshots

#### Testing / Reviewing

Ensure token values are correct for each theme

### CODE REVIEW

File: packages/themes/src/tokens/v11TokenGroup.js [hunk 1]
```diff
@@ -286,6 +286,7 @@ export const icon = TokenGroup.create({
       state: 'disabled',
       name: 'icon-disabled',
     },
+    'icon-inverse',
```
Comment (alisonjoseph) [inline packages/themes/src/tokens/v11TokenGroup.js]: should this be icon-interactive?
```suggestion
    'icon-interactive',
```

File: packages/themes/src/tokens/__tests__/metadata-test.js [hunk 2]
```diff
@@ -877,6 +877,10 @@ test('metadata', () => {
           "name": "icon-disabled",
           "type": "color",
         },
+        Object {
+          "name": "icon-inverse",
```
Comment (alisonjoseph) [inline packages/themes/src/tokens/__tests__/metadata-test.js]:
```suggestion
          "name": "icon-interactive",
```

same with this?

**Review by alisonjoseph — CHANGES_REQUESTED** [review]

File: packages/themes/src/tokens/v11TokenGroup.js (hunk 1 above)
Comment (tw15egan) [reply packages/themes/src/tokens/v11TokenGroup.js]: yeah whoops! 🙃  Thanks for catching

File: packages/themes/src/tokens/__tests__/__snapshots__/v11-test.js.snap [hunk 3]
```diff
@@ -225,6 +226,7 @@ Array [
   "icon-on-color",
   "icon-on-color-disabled",
   "icon-disabled",
+  "icon-inverse",
```
Comment (tw15egan) [inline packages/themes/src/tokens/__tests__/__snapshots__/v11-test.js.snap]:
```suggestion
  "icon-interactive",
```

File: packages/themes/src/tokens/__tests__/__snapshots__/v11-test.js.snap [hunk 4]
```diff
@@ -61,6 +61,7 @@ Array [
...
```
Comment (tw15egan) [inline packages/themes/src/tokens/__tests__/__snapshots__/v11-test.js.snap]:
```suggestion
  "icon-interactive",
```

**Comment by tw15egan** [comment]
@alisonjoseph should be fixed now, thanks for catching that 😅

**Review by alisonjoseph — APPROVED** [review]
🚀

**Review by francinelucca — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Comment by tay1orjones** [comment]
@tw15egan Any idea why the elements preview didn't deploy for this PR? I was going to check it but it's not listed 🤔

**Comment by tw15egan** [comment]
@tay1orjones I just checked a few other PR's and I'm not seeing it on any of them ... 😬

**Comment by tw15egan** [comment]
@thyhmdo this should be ready for review now 👍🏻

**Review by thyhmdo — APPROVED** [review]
looks good. Thanks!

**Pull request merged**